src/gui/torrentcontenttreeview.cpp

```diff
@@ -106,6 +108,21 @@ void TorrentContentTreeView::keyPressEvent(QKeyEvent *event)
     }
 }
 
+void TorrentContentTreeView::mouseDoubleClickEvent(QMouseEvent *event)
+{
+    const QModelIndex index = indexAt(event->pos());
+    if (index.isValid())
+    {
+        QStyleOptionViewItem option;
+        option.initFrom(this);
+        option.rect = visualRect(index);
+        const auto checkRect = style()->subElementRect(QStyle::SE_CheckBoxClickRect, &option, this);
+        if (!checkRect.contains(event->pos()))
+            emit itemDoubleClickedAction(index);
```

Comment:
Wouldn't it be better to continue using `QAbstractItemView::doubleClicked()` signal, but prevent it from being emitted in the wrong case, instead of using this custom signal?

Reply:
the only way I can think to stop the signal is by not calling the QAbstractItemView::mouseDoubleClick event when overriding the same but it seems to do other things https://code.woboq.org/qt5/qtbase/src/widgets/itemviews/qabstractitemview.cpp.html#_ZN17QAbstractItemView21mouseDoubleClickEventEP11QMouseEvent
https://code.woboq.org/qt5/qtbase/src/widgets/itemviews/qtreeview.cpp.html#_ZN9QTreeView21mouseDoubleClickEventEP11QMouseEvent

Reply:
>the only way I can think to stop the signal is by not calling the QAbstractItemView::mouseDoubleClick event when overriding the same

:+1:
>but it seems to do other things

But we want it to just switch the checkbox if it happens within its boundaries, don't we? Do you think that this will prevent any important actions? You can just try it and see what happens.

By the way, one more question. What does doubleclick on folder checkbox do? Does it expand/collapse it or not?

Reply:
> What does doubleclick on folder checkbox do? Does it expand/collapse it or not?

we explicitly disable it. 

>Do you think that this will prevent any important actions?

default implementation selects unselects item based on the type of click. I don't want to handle these actions myself.